pkg/tfbridge/validate_input_types.go

```diff
@@ -157,9 +157,8 @@ func (v *PulumiInputValidator) validatePropertyValue(
 	}
 
 	switch typeSpec.Type {
-	case "bool":
-		// The bridge permits coalescing strings to booleans, hence skip strings.
-		if !propertyValue.IsBool() && !propertyValue.IsString() {
```

Comment:
I've searched the code and apparently over here https://github.com/pulumi/pulumi-terraform-bridge/blob/t0yv0%2Ffix-typechecking-booleans/pkg/tfbridge/schema.go#L431 the bridge is not parsing strings when the slot is of type boolean, at least not yet. Removing `!propertyValue.IsBool()` condition matches this behavior.